frontend/components/flashcard/Flashcard.tsx

```diff
@@ -1,155 +1,268 @@
 import React, { useState } from "react";
-import { ArrowLeft, ArrowRight, X, Edit2, Check } from "react-feather";
+import {
+	ArrowLeft,
+	ArrowRight,
+	X,
+	Edit2,
+	Check,
+	FilePlus,
+	FileMinus,
+} from "react-feather";
 import { Button } from "../input/Button";
+import { InputField } from "../form/InputField";
+import { authedFetch } from "../../util/reauth";
+import { useSWRConfig } from "swr";
+import { toast } from "react-toastify";
+import { Card } from "../../types/Card";
 
-interface CardProps {
+interface FlashcardProps {
 	/**
 	 * unique identifier
 	 */
 	id: string;
 	/**
 	 * Flashcard
 	 */
-	card: { header: string; description: string }[];
+	card: Card;
 	/**
-	 * How many cards are left on the stack
+	 * Cards left to learn
 	 */
-	cardsleft: number;
+	dueCards?: number;
 	/**
-	 * CardSide to show
+	 * Flashcard side to show
 	 */
 	cardSide?: number;
 	/**
 	 * Enables edit mode
 	 */
 	isEdit?: boolean;
+	/**
+	 * Flashcard size
+	 */
+	fullSize?: boolean;
 	/**
 	 * Additional classes
 	 */
 	className?: string;
+	/**
+	 * callback to push rating
+	 */
+	push?: (body: { cardID: string; rating: number }) => void;
 }
 
 /**
  * UI component for displaying flashcards
  */
-export const Card = ({
+export const Flashcard = ({
 	id,
 	card,
-	cardsleft,
+	dueCards,
 	cardSide = 0,
 	isEdit = false,
-	className,
-}: CardProps) => {
-	const [flashCard, setFlashCard] = useState(card);
-	const [tempCard, setTempCard] = useState(flashCard);
-	const [side, setSide] = useState(cardSide % tempCard.length);
-	const [edit, setEdit] = useState(isEdit);
+	fullSize = false,
+	className = "",
+	push,
+}: FlashcardProps) => {
+	const { mutate } = useSWRConfig();
 
-	function editField(field: "header" | "description", value: string) {
-		const card = structuredClone(tempCard);
-		card[side][field] = value;
-		setTempCard(card);
-	}
+	const [tempCard, setTempCard] = useState<Card>(card);
+	const [side, setSide] = useState<number>(cardSide % card.sides?.length);
```

Comment:
2 % undefined => `NaN`

```suggestion
	const [side, setSide] = useState<number>(cardSide % (card.sides?.length || 1));
```

Reply:
fixed in 7b69932b913ca6cf11db44fd1071d49e1ef06ade